hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/lock/OmLockOperation.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.om.lock;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.locks.Lock;
+import org.apache.hadoop.ozone.om.exceptions.OMException;
+import org.apache.hadoop.util.Time;
+import org.apache.ratis.util.function.CheckedBiFunction;
+import org.apache.ratis.util.function.CheckedFunction;
+
+/**
+ * Manage locking of volume, bucket, keys and others.
+ */
+public class OmLockOperation {
+  private static final long LOCK_TIMEOUT_DEFAULT = 10 * 60 * 1000;
+
+  private final WrappedStripedLock keyLocking;
+  private final WrappedStripedLock bucketLocking;
+  private final WrappedStripedLock volumeLocking;
+
+  public OmLockOperation() {
+    keyLocking = new WrappedStripedLock(102400, LOCK_TIMEOUT_DEFAULT, false);
+    bucketLocking = new WrappedStripedLock(1024, LOCK_TIMEOUT_DEFAULT, false);
+    volumeLocking = new WrappedStripedLock(1024, LOCK_TIMEOUT_DEFAULT, false);
+  }
+
+  public OmLockObject lock(OmLockInfo.LockInfoProvider lockInfo) throws IOException {
+    OmLockObject omLockObject = new OmLockObject(lockInfo);
+    long startTime = Time.monotonicNowNanos();
+    try {
+      OmLockInfo.LockLevel level = lockInfo.getLevel();
+      switch (level) {
```

Comment:
When using inheritance, don't use switch-case; see 
- https://stackoverflow.com/questions/31518236/object-oriented-programming-avoid-switch-case-and-if-else-java
- https://softwareengineering.stackexchange.com/questions/347223/switch-vs-polymorphism

Reply:
Yes, above are better to avoid switch. will take up with finalized solution.